### PR TITLE
WIP: Feature/anti grief

### PR DESCRIPTION
**Pull request by Isfirs**
# Changes proposed in this pull request:
- Add protection to a colony
- Restrict access to various actions like
 - detonation
(DONE, NEEDS ENHANCEMENT)
 - mob grief
 - GUIs

This will give a foundation for later usage.

TODOs:
- loggers (DONE)
- debug
- test
- remove WIP status

### CODE REVIEW

File: src/main/java/com/minecolonies/permissions/events/ColonyPermissionEventHandler.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+                    if (rank.ordinal() >= Permissions.Rank.OFFICER.ordinal())
+                    {
+                        return true;
+                    }
+                }
+            }
+        }
+
+        /*
+         * - We are not inside the colony
```
Comment (cltnschlosser) [inline src/main/java/com/minecolonies/permissions/events/ColonyPermissionEventHandler.java]: You should be able to unit test these methods.

Comment (Kostronor) [inline src/main/java/com/minecolonies/permissions/events/ColonyPermissionEventHandler.java]: Agreed, good point

Comment (Isfirs) [inline src/main/java/com/minecolonies/permissions/events/ColonyPermissionEventHandler.java]: Well, I do not have any good experience in unittesting, espacially in this modded context.

Can you take that task?

**Comment by Isfirs** [comment]
I will move this into branch `feature/#5-anti-grief`.

Closing

**Comment by Kostronor** [comment]
Good idea!  @Isfirs

**Pull request closed without merging**